tests/rptest/scale_tests/many_topics_test.py

```diff
@@ -585,9 +662,12 @@ def restart_node(node):
             self.redpanda.restart_nodes(node)
             if safe:
                 self._disable_maintenance_mode(node)
-                self._wait_for_leadership_balanced()
+                self._wait_until_cluster_healthy(include_underreplicated=True)
 
-        for node in self.redpanda.started_nodes():
+        started_nodes = self.redpanda.started_nodes()
+        nodes = started_nodes[:min(max_nodes, len(started_nodes)
```

Comment:
seems like this could be simplified to `started_nodes[:max_nodes or started_nodes]`, though this and the original code are both wrong for max_nodes=0, which is probably fine

Reply:
Completely forgot about `or`. Will switch over.

Reply:
I think I was trying to handle the case where `max_nodes` > `len(started_nodes)`. Something like:
```python
started_nodes[:min(max_nodes or len(started_nodes), len(started_nodes))]
```
would work though.

Reply:
Though now that I checked it looks like in Python you can have `c = a[:b]` where `b > len(a)`  and it'll just set `c == a`. Was thinking it'd raise an exception in that case. You're original suggestion is  the most succinct then.